src/appmetrics.cpp

```diff
@@ -87,7 +87,12 @@ namespace monitorApi {
 
 static std::string toStdString(Local<String> s) {
     char *buf = new char[s->Length() + 1];
-    s->WriteUtf8(buf);
+    #if NODE_VERSION_AT_LEAST(10, 0, 0)
+        Isolate* isolate = v8::Isolate::GetCurrent();
+        s->WriteUtf8(isolate, buf);
+	#else
+        s->WriteUtf8(buf);
+	#endif
```

Comment:
Indentation is kind of off here. Typically the preprocessor lines (e.g. those beginning with `#`) would be in column 0 and the code blocks at the same indentation as if the preprocessor lines were not there.

e.g. 
```c++
static std::string toStdString(Local<String> s) {
    char *buf = new char[s->Length() + 1];
#if NODE_VERSION_AT_LEAST(10, 0, 0)
    Isolate* isolate = v8::Isolate::GetCurrent();
    s->WriteUtf8(isolate, buf);
#else
    s->WriteUtf8(buf);
#endif
```

Reply:
https://github.com/nodejs/nan/blob/master/doc/string_bytes.md#nandecodewrite perhaps? It shouldn't be necessary to have node version ifdefs, at least not for something as common as string conversions.

Reply:
I'm not sure how to implement that Nan function tbh, as I don't see how the function name and arguments are related. And [Nan seems to implement `writeUtf8` using 4 or 5 arguments](https://github.com/nodejs/nan/blob/master/nan.h#L1146)? Happy to implement this if you show me how